xarray/tests/test_backends.py

```diff
@@ -2163,6 +2163,8 @@ def skip_if_not_engine(engine):
         pytest.importorskip(engine)
 
 
+@pytest.mark.filterwarnings("ignore:In xarray version 0.14 `auto_combine` "
+                            "will be deprecated")
```

Comment:
I know it would probably be save to ignore these given that the output of these tests is probably not going to change, but generally we try to update xarray's test suite to avoid our own warnings  rather than filtering them. This helps identify any possible issues with the new warnings, because we can see how pervasive the effect is on our own code.

Reply:
> try to update xarray's test suite to avoid our own warnings rather than filtering them

I don't think that's possible in this case - `auto_combine` will throw a `FutureWarning` whatever arguments you give it, but we still want to test it. This is only temporary though - once `auto_combine` is fully deprecated all the tests which filter warnings will be deleted.

Reply:
I think this test uses `open_mfdataset`, so we could probably just add the keyword argument that triggers the new code path. Or we could add an explicit `pytest.assert_warns()` check, which would fail if the warning goes away. Either way would be preferable, because it means we won't have to remember to update this later.

Reply:
> I think this test uses open_mfdataset, so we could probably just add the keyword argument that triggers the new code path.

We could do that, but then the old code path wouldn't be tested anymore?

> Or we could add an explicit pytest.assert_warns() check, which would fail if the warning goes away.

Happy to do that too

Reply:
I think the underlying `auto_combine` has tests somewhere?

Otherwise, or if you're not sure those are sufficient, we can do the explicit `assert_warns`. It's definitely good to check that warnings are actually issued, but let's keep them in the minimum number of places that code will just get deleted later.

Reply:
> I think the underlying auto_combine has tests somewhere?

Actually yeah you're right, that code path will still be sufficiently tested.

> good to check that warnings are actually issued

Okay so just change the tests in `test_backends.py` to circumvent the deprecation warnings? As we are still checking that warnings are raised in `test_combine.py`.

Reply:
> Okay so just change the tests in `test_backends.py` to circumvent the deprecation warnings? As we are still checking that warnings are raised in `test_combine.py`.

Yes, but let's keep one test using `open_mfdataset()` that checks for the custom warning issued there.

Reply:
> Okay so just change the tests in test_backends.py to circumvent the deprecation warnings?

Done.

> Yes, but let's keep one test using open_mfdataset() that checks for the custom warning issued there.

Turns out I already wrote these - see `TestOpenMFDataSetDeprecation`. So the only test changes needed when completing this deprecation cycle are deleting the test classes `TestOpenMFDataSetDeprecation`, `TestAutoCombineOldAPI` and `TestAutoCombineDeprecation`. (Also you could optionally remove some `combine` arguments to `open_mfdataset` as there will be a new default.)